src/Cache/Internal/ScsDataClient.php

```diff
@@ -1513,6 +1526,115 @@ function () use ($call): SetRemoveElementResponse {
+            validateCacheName($cacheName);
+            validateSortedSetName($sortedSetName);
+            validateValueName($value);
+            $ttlMillis = $this->ttlToMillis($collectionTtl->getTtl());
+            validateTtl($ttlMillis);
+            $element = new _SortedSetElement([
+                'value' => $value,
+                'score' => $score,
+            ]);
+            $sortedSetPutElementRequest = new _SortedSetPutRequest();
+            $sortedSetPutElementRequest->setSetName($sortedSetName);
+            $sortedSetPutElementRequest->setRefreshTtl($collectionTtl->getRefreshTtl());
+            $sortedSetPutElementRequest->setTtlMilliseconds($ttlMillis);
+            $sortedSetPutElementRequest->setElements([$element]);
+            $call = $this->grpcManager->client->SortedSetPut(
+                $sortedSetPutElementRequest,
+                ["cache" => [$cacheName]],
+                ["timeout" => $this->timeout],
+            );
+        } catch (SdkError $e) {
+            return ResponseFuture::createResolved(new SortedSetPutElementError($e));
+        } catch (Exception $e) {
+            return ResponseFuture::createResolved(new SortedSetPutElementError(new UnknownError($e->getMessage(), 0, $e)));
+        }
+
+        return ResponseFuture::createPending(
+            function () use ($call): SortedSetPutElementResponse {
+                try {
+                    $this->processCall($call);
+
+                    return new SortedSetPutElementSuccess();
+                } catch (SdkError $e) {
+                    return new SortedSetPutElementError($e);
+                } catch (Exception $e) {
+                    return new SortedSetPutElementError(new UnknownError($e->getMessage(), 0, $e));
+                }
+            }
+        );
+    }
+
+    /**
+     * @return ResponseFuture<SortedSetFetchResponse>
+     */
+    public function sortedSetFetchByRank(string $cacheName, string $sortedSetName, ?int $startRank = 0, ?int $endRank = null, ?int $order = SORT_ASC): ResponseFuture
+    {
+        try {
+            validateCacheName($cacheName);
+            validateSortedSetName($sortedSetName);
+            validateSortedSetRanks($startRank, $endRank);
+
+            $sortedSetFetchRequest = new _SortedSetFetchRequest();
+            $sortedSetFetchRequest->setSetName($sortedSetName);
+            $sortedSetFetchRequest->setWithScores(true);
+
+            $byIndex = new _SortedSetFetchRequest\_ByIndex();
+            if ($startRank) {
+                $byIndex->setInclusiveStartIndex($startRank);
+            } else {
+                $byIndex->setUnboundedStart(new _Unbounded());
+            }
+            if ($endRank) {
+                $byIndex->setExclusiveEndIndex($endRank);
+            } else {
+                $byIndex->setUnboundedEnd(new _Unbounded());
+            }
+            $sortedSetFetchRequest->setByIndex($byIndex);
+
+            if (!$order || $order >= SORT_ASC) {
```

Comment:
What does >= mean here?

Reply:
order is just an integer. SORT_DESC is 3 and SORT_ASC is 4. So, if the user supplies 3 or lower, we choose descending, or if the supply 4 or higher we choose ascending. I wanted to cover the cases where they gave integers other than SORT_ASC or SORT_DESC.

Reply:
I should change that to `is_null($order)`, since I think 0 will evaluate to ascending with the current logic.

Reply:
Maybe in a separate PR (due to CI jobs 😢 )